docs/upgrade.html

```diff
@@ -66,12 +66,23 @@
         individual messages is only reduced by the overhead of the batch format. This similarly affects the
         producer's <code>batch.size</code> configuration.</li>
     <li>GC log rotation is enabled by default, see KAFKA-3754 for details.</li>
+    <li>Message protocol introduces support for headers as an array of Header with uftf8 <code>key</code> and byte[] <code>value</code>. 
+        Header's will be available once the broker is upgraded to the latest protocol. Older clients will not be able to access the headers as they don't exist in previous protocols as such the broker removes the header on protocol downgrade.</li>
+    <li>ProducerRecord and ConsumerRecord client API's expose the new Headers API via <code>Headers headers()</code> method call. 
```

Comment:
To clarify, `upgrade.html` usually only includes changes with a compatibility impact. At the moment, items that should be part of the release notes are added to the release plan:

https://cwiki.apache.org/confluence/display/KAFKA/Release+Plan+0.11.0.0

Actual documentation of how Headers work, etc. should be added to the Javadoc and potentially to the Kafka documentation (see docs folder).